### PR TITLE
Pin meta.yaml to llvmlite 0.38 series

### PR DESCRIPTION
**Pull request by sklam**
as titled

### CODE REVIEW

**Comment by sklam** [comment]
Without this, llvmlite 0.39.0dev0 can be installed with numba 0.55 causing `numba.tests.test_entrypoints` and `numba.tests.test_typedlist.TestTypedList.test_repr_long_list_ipython` to fail. Both failures are caused by llvmpy deprecation warnings.

File: buildscripts/condarecipe.local/meta.yaml [hunk 1]
```diff
@@ -35,7 +35,7 @@ requirements:
     - numpy
     - setuptools
     # On channel https://anaconda.org/numba/
-    - llvmlite >=0.38.0rc1,<0.39
+    - llvmlite 0.38.*
```
[the remark this answers is not in the document]
Comment (sklam) [reply buildscripts/condarecipe.local/meta.yaml:38]: The <0.39 is picking up dev builds on 0.39 though. That's the main reason to change.

[the remark this answers is not in the document]
Comment (sklam) [reply buildscripts/condarecipe.local/meta.yaml:38]: `0.38.*` would only pick up 0.38 rc and dev builds. And one will have to be quite creative to do that given the `numba/label/dev` channel is the only place to get them.

[the remark this answers is not in the document]
Comment (stuartarchibald) [reply buildscripts/condarecipe.local/meta.yaml:38]: Thanks for the explanation. From OOB discussion, it was elected to leave this as-is. `0.38.*` should be sufficient.

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch!

**Pull request merged**